src/react-loadable-component.js

```diff
@@ -20,6 +20,9 @@ export default function ReactLoadableComponent(name, resolve, files = []) {
     }
 
     render() {
+      if (this.state.error) {
+        throw this.state.error;
```

Comment:
Can we maybe do this by adding an option/flag to `ReactLoadableComponent`?

Reply:
What will be the component behavior when the option is opted out?

Reply:
The same as it is now, although again, this makes sense only if we cover the `resolvedData` null check with this flag.

The point is both these changes can have a subtle effect on various cases. I'm not sure it's always correct to throw if resolve fails, and I'm not sure it's correct to "swallow" the null case for existing components, without throwing.

Reply:
Okay, I will re-phrase, while it might be "correct" I mean that it's hard to understand the effect on existing components.

Reply:
Analyzing the code paths, I think this the `null` check + error bubbling should cover all use cases.

Giving that today the bubbling  happens by `null` error:
* `null` checks prevents invalid access to `resolvedData`
* `setState({error: err});` allows to bubble up the error, since now it's not bubbled by the bug.

Reply:
Sorry, missed this reply, I believe that if `filesAppenderPromise` fails, then the component might still be rendered, but after this change it throws an error. In the case of `ReactLoadableComponent`, `filesAppenderPromise` could be considered a non-critical error.

But I might be missing something again :), But previously the error was thrown from `componentDidMount` but your change moves it to be the same but from `render`, so why do you need that change actually? Why is throwing in `componentDidMount` a problem but no in `render`?

Reply:
But I think if we ignore the `filesAppenderPromise` and decide we should also fail on it then this is correct.

Reply:
`filesAppenderPromise` has the same catch handler as `resolvePromise`

Reply:
you mean `resolveData` will still be `null` in that case? True, so the only change here overall is moving the `throw` from `componentDidMount` to `render`?

Reply:
Yes